src/utilities/logging/Engine.ts

```diff
@@ -0,0 +1,60 @@
+import bunyan from 'react-native-bunyan';
+import { Action } from '.';
+import { Transport } from './Transport';
+// Logging engine for the Logger class which implements the logic
+// of creating a log format and hands off the log to a transport
+// object.
+//
+// interface LoggingEngine {
+//     trace: (text: string) => void
+//     debug: (text: string) => void
+//     info: (text: string) => void
+//     warn: (text: string) => void
+//     error: (text: string) => void
+//     fatal: (text: string) => void
+//     setLogLevel(transportKey: string, newLevel: int) => void
+// }
+
+export class BunyanLoggingEngine {
+  bunyan: null | any = null;
```

Comment:
didn't try very hard to correctly type the bunyan as it is only referenced in the `LoggerService`